### DISCUSSION TITLE
Porting SBFspot to macOS silicon

### DISCUSSION THREAD

**Discussion by shieling** [SBFspot/SBFspot#585] [Q&A] [answered]
I have successfully installed SBFspot on my Ubuntu laptop and now wish to install it on my desktop Mac Studio. I am not too familiar with C++ but would like to try using Xcode IDE. The SBFspot I would like to install is without sql database. The Mac Studio already has boost and bluetooth installed, but obviously not in the same locations as on Ubuntu. I have imported all the source code into Xcode, but want to know how to start building the application using the already installed components. Any help would be appreciated.

**Comment by sbf-** [comment]
Maybe @mincequi can give you some hints?

**Comment by shieling** [comment]
I have tried compiling the SBFspot in Xcode but having real trouble with boost. Many of the boost components can’t be found. I tried copying the boost .hpp files to a boost subdirectory, but this doesn’t seem to help.

**Comment by mincequi** [comment]
Yes, i have my own fork which compiles and runs under macOS: https://github.com/mincequi/SBFspot

If you want to compile this, you probably have to deal with homebrew. Is this known to you?
Besides some requirements, you might need some further twiddling with the source code and build scripts.

However, it might largely depend on your requirements, if SBFspot is really the right thing for you (and a macOS installation).

**Comment by shieling** [comment, marked as answer]
After a series of very useful conversations with mincequi, I will close this question and continue with my attempt to portr the SBFspot to my MAC Studio M1.

**Reply by deancs** [reply]
Hello, were you successful in getting sbfspot running on Mac? If so would you share the code?.

**Reply by shieling** [reply]
Unfortunately I didn’t manage to make it work, so gave up and installed it on a Raspberry Pi installed with Ubuntu. This suited me better in the end because I leave the RP on 24/7, while my Mac is only on during the day. The RP uses a cron to update the MAC when it is on. The cron also updates my personal website with both the SMA data and my weather station - www.theshieling.net.
Thanks for the interest.

**Comment by deancs** [comment]
OK, thanks for the explanation - its appreciated!
